cron_ext/extension.py

```diff
@@ -0,0 +1,232 @@
+from __future__ import annotations
+
+import json
+import os
+import re
+import stat
+import subprocess
+import sys
+from contextlib import suppress
+from hashlib import sha256
+from pathlib import Path
+from typing import Callable, Iterable
+
+import structlog
+from cached_property import cached_property
+from meltano.edk import models
+from meltano.edk.extension import ExtensionBase
+
+from cron_ext import APP_NAME
+
+log = structlog.get_logger(APP_NAME)
+
+
+def run_subprocess(
+    args: tuple[str],
+    error_message: str,
+    exit_on_nonzero_returncode: bool = True,
+) -> subprocess.CompletedProcess:
+    try:
+        proc = subprocess.run(
+            args,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            text=True,
+        )
+    except Exception:
+        log.exception(error_message)
+        sys.exit(1)
+    if proc.returncode:
+        if exit_on_nonzero_returncode:
+            log.critical(error_message)
+            sys.exit(proc.returncode)
+        log.error(error_message)
+    return proc
+
+
+class Cron(ExtensionBase):
+    full_line_comment_pattern = re.compile(r"^\s*#.*$")
+    entry_pattern = re.compile(r"(?i)^.+?'(?P<path>(?:/.*)*/(?P<name>.*)\.sh)'.*$")
+
+    def invoke(self, *args, **kwargs):
+        """Invoke the underlying cli, that is being wrapped by this extension."""
+        raise NotImplementedError
+
+    @cached_property
+    def meltano_project_dir(self) -> Path:
+        return Path(os.environ["MELTANO_PROJECT_ROOT"]).resolve()
+
+    @cached_property
+    def cron_ext_dir(self) -> Path:
+        path = self.meltano_project_dir / ".meltano" / "cron-ext"
```

Comment:
nit: I don't think an env var is injected by meltano for this, but usually plugins will write to their `run dir`  by default. e.g. https://github.com/meltano/meltano/blob/main/src/meltano/core/plugin/airflow.py#L30

```suggestion
        path = self.meltano_project_dir / ".meltano" /  "run" / "cron-ext"
```